src/java.base/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -252,7 +252,7 @@ public SegmentScope session() {
          * Create a binding context from given scope. The resulting context will throw when
          * the context's allocator is accessed.
          */
-        public static Context ofSession() {
+        public static Context ofScope() {
```

Comment:
I think `ofArena` seems more apt here?

Reply:
More generally, I think that now that we can subclass Arena, Binding.Context can just become a subclass of Arena. Then we might have several implementations, some of which throw on allocate(), some other will throw on scope() and close(). Seems a more direct approach.

Reply:
Let's do that in a separate PR.